.github/workflows/deploy.yaml

```diff
@@ -0,0 +1,23 @@
+name: Deploy
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    branches:
+      - "*"
```

Comment:
After testing this out, I would just remove the `pull_request` rule entirely, right? We only need to publish the docs on any merge to `main`.

The idea of a preview URL is super cool but we'd need something like Netlify for that that to work (I don't think GitHub Pages could do it).

Reply:
Yup will do.